quantum/nodes/cnot-gate/cnot-gate.js

```diff
@@ -0,0 +1,96 @@
+    const node = this;
+
+    node.on('input', async function(msg, send, done) {
+      // Throw a connection error if:
+      // - The user connects it to a node that is not from the quantum library
+      // - The user does not input a qubit object in the node
+      // - The user chooses to use registers but does not initiate them
+      // - The user does not input 2 qubits.
+      if (msg.topic !== 'Quantum Circuit') {
+        throw new Error(
+            'The CNot Gate must be connected to nodes from the quantum library only.',
+        );
+      } else if (typeof(msg.payload.register) === 'undefined' && typeof(msg.payload.qubit) === 'undefined') {
+        throw new Error(
+            'The CNot Gate nodes must receive qubits objects as inputs.\n' +
+            'Please use "Quantum Circuit" & "Quantum Register" nodes to generate qubits objects.',
+        );
+      } else if (typeof(msg.payload.qubit) === 'undefined') {
+        throw new Error(
+            'If "Registers & Bits" was selected in the "Quantum Circuit" node, please make use of register nodes.',
+        );
+      } else if (node.qubits.length !== 2) {
+        throw new Error(
+            'The CNot Gate requires exactly 2 input qubits to be used.',
+        );
+      }
+
+      // Store all the qubit objects received as input into the node.qubits array
+      node.qubits.push(msg);
+
+      // If all qubits have arrived, we first reorder the node.qubits array for output consistency
+      if (node.qubits.length === 2) {
+        node.qubits.sort(function compare(a, b) {
+          if (typeof (a.payload.register) !== 'undefined') {
+            const regA = parseInt(a.payload.registerVar.slice(2));
+            const regB = parseInt(b.payload.registerVar.slice(2));
+            if (regA < regB) return -1;
+            else if (regA > regB) return 1;
+            else {
+              if (a.payload.qubit < b.payload.qubit) return -1;
+              else return 1;
+            }
+          } else {
+            if (a.payload.qubit < b.payload.qubit) return -1;
+            else return 1;
+          }
+        });
+
+        if (targetPosition === 'upper') {
+          let targetQubit = node.qubits[0];
+          let controlQubit = node.qubits[1];
+        } else {
+          let targetQubit = node.qubits[1];
+          let controlQubit = node.qubits[0];
+        }
+
+        // Generate the corresponding cnot-gate Qiskit script
+        if (typeof msg.payload.register === 'undefined') {
+          let cnotScript = util.format(
+              snippets.CNOT_GATE,
+              controlQubit.qubit.toString(),
+              targetQubit.qubit.toString(),
+          );
+        } else {
+          let cnotScript = util.format(
+              snippets.CNOT_GATE,
+              controlQubit.registerVar + '[' + controlQubit.qubit.toString() + ']',
+              targetQubit.registerVar + '[' + targetQubit.qubit.toString() + ']',
+          );
+        }
+
```

Comment:
The qubits received as input are msg with a `payload` property.

To access the qubit data, you need to use: `controlQubit.payload.qubit.toString()`

Reply:
yeah, you're right, thank you Theo